samples/nrf9160/http_update/common/src/update.c

```diff
@@ -82,22 +76,22 @@ int cert_provision(void)
 
 static int led_init(int num_leds)
 {
-	const struct device *dev;
-
-	dev = device_get_binding(LED_PORT);
-	if (dev == 0) {
-		printk("Nordic nRF GPIO driver was not found!\n");
-		return 1;
-	}
-
 	switch (num_leds) {
 	case 0:
 		return 0;
 	case 2:
-		gpio_pin_configure(dev, LED1_PIN, LED1_FLAGS);
-		/* Fallthrough */
+		if (!device_is_ready(led1.port)) {
+			return -ENODEV;
+		}
+
+		gpio_pin_configure_dt(&led1, GPIO_OUTPUT_ACTIVE);
+		__fallthrough;
 	case 1:
-		gpio_pin_configure(dev, LED0_PIN, LED0_FLAGS);
+		if (!device_is_ready(led0.port)) {
+			return -ENODEV;
+		}
+
+		gpio_pin_configure_dt(&led0, GPIO_OUTPUT_ACTIVE);
```

Comment:
Why the LEDs are initialized in active state? It seems that the original code did not do so.

Reply:
we had before:

```
#define LED0_FLAGS (GPIO_OUTPUT_ACTIVE | DT_GPIO_FLAGS(DT_ALIAS(led0), gpios))
#define LED1_FLAGS (GPIO_OUTPUT_ACTIVE | DT_GPIO_FLAGS(DT_ALIAS(led1), gpios))
```

Reply:
Ah, right. I missed that. I somehow didn't expect such modifications in definitions of `LEDx_FLAGS`.